stateless-asks/program/src/processor.rs

```diff
@@ -0,0 +1,146 @@
+//! Program state processor
+
+use crate::instruction::DepositType;
+use {
+    borsh::{BorshDeserialize, BorshSerialize},
+    num_traits::FromPrimitive,
+    solana_program::{
+        account_info::next_account_info,
+        account_info::AccountInfo,
+        borsh::try_from_slice_unchecked,
+        clock::{Clock, Epoch},
+        decode_error::DecodeError,
+        entrypoint::ProgramResult,
+        msg,
+        program::{invoke, invoke_signed},
+        program_error::PrintProgramError,
+        program_error::ProgramError,
+        program_pack::Pack,
+        pubkey::Pubkey,
+        rent::Rent,
+        stake_history::StakeHistory,
+        system_instruction, system_program,
+        sysvar::Sysvar,
+    },
+    spl_token::state::Mint,
+};
+
+/// Program state handler.
+pub struct Processor {}
+impl Processor {
+    /// Issue a spl_token `Transfer` instruction.
+    #[allow(clippy::too_many_arguments)]
+    fn token_transfer<'a>(
+        token_program: AccountInfo<'a>,
+        source: AccountInfo<'a>,
+        destination: AccountInfo<'a>,
+        authority: AccountInfo<'a>,
+        seeds: &[&[u8]],
+        amount: u64,
+    ) -> Result<(), ProgramError> {
+        let ix = spl_token::instruction::transfer(
+            token_program.key,
+            source.key,
+            destination.key,
+            authority.key,
+            &[],
+            amount,
+        )?;
+        invoke_signed(&ix, &[source, destination, authority, token_program], seeds)
+    }
+
+    /// Processes `Initialize` instruction.
+    fn process_accept_offer(
+        program_id: &Pubkey,
+        accounts: &[AccountInfo],
+        maker_size: u64,
+        taker_size: u64,
+        bump_seed: u8,
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+
+        let maker_src_account = next_account_info(account_info_iter)?;
+        let maker_dst_account = next_account_info(account_info_iter)?;
+
+        let taker_src_account = next_account_info(account_info_iter)?;
+        let taker_dst_account = next_account_info(account_info_iter)?;
+
+        let auth_maker_account = next_account_info(account_info_iter)?;
+        let auth_taker_account = next_account_info(account_info_iter)?;
+
+        let token_program_info = next_account_info(account_info_iter)?;
+
+        let transfer_authority = next_account_info(account_info_iter)?;
```

Comment:
How would a client program go about finding these PDA's? The completely stateless idea is cool, but it becomes difficult to discover any of these accounts right? You might be able to see who the delegate is on a token account, but even then, how would you know the seeds?